app/models/solidus_paypal_commerce_platform/paypal_order.rb

```diff
@@ -121,6 +122,18 @@ def price(amount)
       }
     end
 
+    def application_context
+      {
+        shipping_preference: require_shipping? ? 'SET_PROVIDED_ADDRESS' : 'NO_SHIPPING'
+      }
+    end
+
+    def require_shipping?
+      step_names = @order ? @order.checkout_steps : ::Spree::Order.checkout_steps.keys
```

Comment:
Hi @retsef 👋 
`@order.checkout_steps` returns strings
`::Spree::Order.checkout_steps.keys` returns symbols

Meaning `step_names.include? 'delivery'` will always return false for the latter.

It would also be nice to move `step_names` to another private method:
```ruby
def step_names
  @order ? @order.checkout_steps.map(&:to_sym) : ::Spree::Order.checkout_steps.keys
end
```
And then in another PR we benefit from your method 🙂 
```ruby
    def javascript_sdk_url(order: nil, currency: nil)
      ...
      commit_immediately = step_names.include? "confirm"
```
This would then fix issue #150.